synapse/lib/stormtypes.py

```diff
@@ -915,6 +919,12 @@ def __init__(self, valu, path=None):
     def __len__(self):
         return len(self.valu)
 
+    def __bool__(self):
+        return bool(self.valu)
+
+    def __str__(self):
+        return self.valu.decode()
```

Comment:
This is unsafe and should not be used to str-ify bytes. We have no gaurante that the data held in self.valu is valid utf8 bytes. We'll throw UnicodeDecodeError's into the runtime.

Reply:
I think it's safe to assume a default encoding of utf8 ( for explicit conversion they can use the decode() method directly ) for an implicit tostr() cast.  I'll catch Unicode exception and wrap it into a badcast.